Functions/Quantification/xASL_quant_HadamardDecoding.m

```diff
@@ -280,19 +264,21 @@
 % And for decoding we want to swap the dimensions such that the Hadamard blocks (==PLDs) are the innermost dimension:
 % TE1/PLD1,TE1/PLD2,TE1/PLD3,TE1/PLD4...TE2/PLD1,TE2/PLD2,TE2/PLD3,TE2/PLD4 (blocks in the first dimension, TEs after)
 
-% This is the original order
-vectorReorderEncoded = 1:nEncodedVolumes;
-
-% Shape to a matrix with TEs first and all the rest later
-vectorReorderEncoded = reshape(vectorReorderEncoded, nInnerRepetitions, nEncodedVolumes/nInnerRepetitions);
+if nInnerRepetitions > 1 % Reordering is only needed when having inner repetitions
+	% This is the original order
+	vectorReorderEncoded = 1:nEncodedVolumes;
 
-% Switch the two dimensions and create a row vector again
-vectorReorderEncoded = reshape(vectorReorderEncoded', 1, nEncodedVolumes);
-
-% Reorder the data
-imEncodedReordered = imEncoded(:,:,:,vectorReorderEncoded);
+	% Shape to a matrix with TEs first and all the rest later
+	vectorReorderEncoded = reshape(vectorReorderEncoded, nInnerRepetitions, nEncodedVolumes/nInnerRepetitions);
 
+	% Switch the two dimensions and create a row vector again
+	vectorReorderEncoded = reshape(vectorReorderEncoded', 1, nEncodedVolumes);
 
+	% Reorder the data
+	imEncodedReordered = imEncoded(:,:,:,vectorReorderEncoded);
```

Comment:
Why don't you keep using the variable imEncoded? Now you introduce a new variable for no reason.

Reply:
You have to check the latest commits, not the middle one. This has been corrected already.